[Issue] rocprofiler-compute test_profile_pc_sampling Fails in CI Due to Unsupported PC Sampling on CI Agents

### Summary:
Recent CI runs are failing due to an error in the PC sampling test. The test test_profile_pc_sampling (in tests/test_profile_general.py, specifically test_pc_sampling_stochastic) fails with the following error:

rocm-systems Bump PR: https://github.com/ROCm/TheRock/pull/3984

### Tests Failing:
```
test_profile_pc_sampling 
```

### log:
```
'test_pc_sampling_stochastic_output', 'ROCPROF_OUTPUT_FILE_NAME': 'ps_file', 'ROCPROFILER_PC_SAMPLING_BETA_ENABLED': '1', 'ROCPROF_PC_SAMPLING_UNIT': 'cycles', 'ROCPROF_PC_SAMPLING_INTERVAL': '1048576', 'ROCPROF_PC_SAMPLING_METHOD': 'stochastic'}
2026-03-18T05:11:12.7149331Z 11:   [34mDEBUG[0m pc sampling rocprof sdk user provided command: ['./tests/mat_mul_max']
2026-03-18T05:11:12.7726922Z 11:    [32mINFO[0m    |-> [rocprofiler-sdk] W20260318 05:11:12.768270 140560788107328 agent.cpp:470] Attempt to enable hip visiblity for agent-2 which is not visible to HSA (ROCR)
2026-03-18T05:11:12.8402803Z 11:    [32mINFO[0m    |-> [rocprofiler-sdk] E20260318 05:11:12.839683 140560788107328 tool.cpp:2005] Given PC sampling configuration is not supported on any of the agents
2026-03-18T05:11:12.8654751Z 11:   [31mERROR[0m [31mPC sampling failed.[0m
2026-03-18T05:11:12.8979918Z 11: F
2026-03-18T05:11:12.8980085Z 11: 
2026-03-18T05:11:12.8980310Z 11: =================================== FAILURES ===================================
2026-03-18T05:11:12.8980616Z 11: _________________________ test_pc_sampling_stochastic __________________________
2026-03-18T05:11:12.8980822Z 11: 
2026-03-18T05:11:12.8981044Z 11: binary_handler_profile_rocprof_compute = <function binary_handler_profile_rocprof_compute.<locals>._handler at 0x7fd7f7c620c0>
2026-03-18T05:11:12.8981308Z 11: 
2026-03-18T05:11:12.8981402Z 11:     @pytest.mark.pc_sampling
2026-03-18T05:11:12.8981577Z 11:     def test_pc_sampling_stochastic(binary_handler_profile_rocprof_compute):
2026-03-18T05:11:12.8981798Z 11:         if soc in ("MI100") or soc in ("MI200"):
2026-03-18T05:11:12.8981943Z 11:             assert True
2026-03-18T05:11:12.8982044Z 11:             return
2026-03-18T05:11:12.8982143Z 11:     
2026-03-18T05:11:12.8982237Z 11:         options = [
2026-03-18T05:11:12.8982346Z 11:             "--block",
2026-03-18T05:11:12.8982443Z 11:             "21",
2026-03-18T05:11:12.8982567Z 11:             "--pc-sampling-method",
2026-03-18T05:11:12.8982695Z 11:             "stochastic",
2026-03-18T05:11:12.8982808Z 11:             "--pc-sampling-interval",
2026-03-18T05:11:12.8982930Z 11:             "1048576",
2026-03-18T05:11:12.8983024Z 11:         ]
2026-03-18T05:11:12.8983293Z 11:         workload_dir = test_utils.get_output_dir()
2026-03-18T05:11:12.8983449Z 11: >       _ = binary_handler_profile_rocprof_compute(
2026-03-18T05:11:12.8983606Z 11:             config,
2026-03-18T05:11:12.8983709Z 11:             workload_dir,
2026-03-18T05:11:12.8983818Z 11:             options,
2026-03-18T05:11:12.8983930Z 11:             check_success=True,
2026-03-18T05:11:12.8984071Z 11:             roof=False,
2026-03-18T05:11:12.8984234Z 11:             app_name="app_mat_mul_max",
2026-03-18T05:11:12.8984404Z 11:         )
2026-03-18T05:11:12.8984483Z 11: 
2026-03-18T05:11:12.8984577Z 11: tests/test_profile_general.py:2583: 
```

--- a/projects/rocprofiler-compute/tests/test_pc_sampling.py
+++ b/projects/rocprofiler-compute/tests/test_pc_sampling.py
@@ -61,6 +61,14 @@ PC_SAMPLING_STOCHASTIC_FILES = sorted([
 ])
 
 
+def is_pc_sampling_not_supported(output):
+    """
+    To be called with the stdout + stderr after profiling.
+    Check whether profiling output said PC sampling is not supported on the machine
+    """
+    return "Given PC sampling configuration is not supported" in output
+
+
 def test_pc_sampling_host_trap(binary_handler_profile_rocprof_compute):
     """
     Test that PC sampling works with --block 21 and --pc-sampling-method host_trap.
@@ -114,15 +122,21 @@ def test_pc_sampling_stochastic(binary_handler_profile_rocprof_compute):
 
     workload_dir = test_utils.get_output_dir()
 
-    _ = binary_handler_profile_rocprof_compute(
+    code, stdout, stderr = binary_handler_profile_rocprof_compute(
         config,
         workload_dir,
         options,
-        check_success=True,
+        check_success=False,
+        capture_output=True,
         roof=False,
         app_name="app_mat_mul_max",
     )
 
+    output = f"{stdout}\n{stderr}"
+    if is_pc_sampling_not_supported(output):
+        pytest.skip("PC sampling is not supported")
+
+    assert code == 0
     file_dict = test_utils.check_non_pmc_files(workload_dir, num_devices, 1)
     assert sorted(list(file_dict.keys())) == sorted(PC_SAMPLING_STOCHASTIC_FILES)
 

--- a/projects/rocprofiler-compute/tests/test_pc_sampling.py
+++ b/projects/rocprofiler-compute/tests/test_pc_sampling.py
@@ -134,6 +134,7 @@ def test_pc_sampling_stochastic(binary_handler_profile_rocprof_compute):
 
     output = f"{stdout}\n{stderr}"
     if is_pc_sampling_not_supported(output):
+        test_utils.clean_output_dir(config["cleanup"], workload_dir)
         pytest.skip("PC sampling is not supported")
 
     assert code == 0

